Started the Wrapper

TheTrueProblematic opened the pull request:
Started a wrapper and got it working. Testing is not built, but the wrapper itself is largely made. 

Part of #18

TheTrueProblematic committed:
GitIgnore Fix

TheTrueProblematic committed:
First Attempt at wrapper

TheTrueProblematic committed:
First full attempt at wrapping but it seems like this is not the way to go

TheTrueProblematic committed:
Quick Gihub Actions Change

TheTrueProblematic committed:
Restructure

TheTrueProblematic committed:
Wrapper Makefiles, Plus Small updates to c files

TheTrueProblematic committed:
Updated Actions To Make Wrapper

TheTrueProblematic committed:
Moved make.sh

TheTrueProblematic committed:
Merge branch 'mwm-wrapperstart' of https://github.com/edhartnett/ncglm into mwm-wrapperstart

TheTrueProblematic committed:
Fixed github actions?

TheTrueProblematic committed:
Removed First Attempt

TheTrueProblematic committed:
Merge branch 'mwm-wrapperstart' of https://github.com/edhartnett/ncglm into mwm-wrapperstart

TheTrueProblematic committed:
Wrapper it self is mostly done

TheTrueProblematic committed:
Just Some Formatting Changes

edhartnett commented on psrc/ncglm.py:
```diff
@@ -1,3 +1,32 @@
+import ctypes
+
+# Creating Libraries
+glm_group = ctypes.CDLL("../src/glm_group.so")
```
OK, I'm starting to understand what you did here, and it's impressive. ;-) 

I see that what you want is to find the .so file for ncglm.

Let's start by taking all the Makefiles out of this PR, and also all the code changes where you added "include" to the include line, like this:
#include "../include/ncglm.h"

Leave the psrc and ptest changes.

I know this will break CI, but make those changes on your branch, and do a push, and then we'll go over the next step.

TheTrueProblematic committed:
Several Large Changes.
Worked on what we talked about last week and broke down a few of the files as well as the makefile restructuring. Still need to convert library to .la

TheTrueProblematic committed:
Prepare for new make system with workflows

edhartnett commented on Makefile.am:
```diff
@@ -18,6 +18,6 @@ FSRC = fsrc
 FTEST = ftest
 endif
 
```
Add a section like the one for Fortran. Like this:

```
# Does the user want the Python API?
if BUILD_PYTHON
PSRC = psrc
PTEST = ptest
endif
```

edhartnett replied:
Also go into the configure.ac file, find the section for enable-fortran, and copy it, changing fortran to python...

edhartnett commented on make.sh:
```diff
@@ -0,0 +1,4 @@
+make -f psrc/Makefiles/event_Makefile
```
Remove this file.

edhartnett commented on src/glm_event_make:
```diff
@@ -0,0 +1,5 @@
+CFLAGS =  $(CXXFLAGS_COMMON) $(CFLAGS_PERSONAL) $(CADD) -shared -fPIC
```
Remove this file.

TheTrueProblematic committed:
Deleted make.sh

TheTrueProblematic committed:
removed glm_event_make

TheTrueProblematic committed:
Added makefile changes

edhartnett commented:
Add something like this to configure.ac, but for python:

```
# Does the user want to enable Fortran library?
AC_MSG_CHECKING([whether Fortran library should be built])
AC_ARG_ENABLE([fortran],
              [AS_HELP_STRING([--enable-fortran],
                              [build the ncglm Fortran library.])])
test "x$enable_fortran" = xyes || enable_fortran=no
AC_MSG_RESULT([$enable_fortran])
AM_CONDITIONAL(BUILD_FORTRAN, [test "x$enable_fortran" = xyes])
```

TheTrueProblematic committed:
Is this the type of change you wanted for the configure.ac file?

edhartnett commented on psrc/ncglm.py:
```diff
@@ -1,3 +1,32 @@
+
+# Creating Libraries
+glm_group = ctypes.CDLL("../src/glm_group.so")
+glm_event = ctypes.CDLL("../src/glm_event.so")
```
Now change this to point to the one .so, libncglm.so.

TheTrueProblematic committed:
fix to use the libncglm.so file that still needs to be made from libncglm.la at this point

TheTrueProblematic committed:
fix to use the libncglm.so file that still needs to be made from libncglm.la at this point

edhartnett commented:
Your configure.ac looks good, and you are now passing CI, which is good.

(Except travis. I will take a look...)

Python code does not have to be compiled, so all we have to do is make sure the file ends up in the tarball distribution. (https://www.gnu.org/software/automake/manual/html_node/Dist.html). We need an EXTRA_DIST = filename in the Makefile.am.

In the ptest directory we need to run the tests. For that, we need a TESTS = testname in the Makefile.am (this is documented, in horrifying complexity, here https://www.gnu.org/software/automake/manual/html_node/Parallel-Test-Harness.html. But it's just gong to be one line: TESTS = script.py)

edhartnett commented:
You need to resolve the confilcts in this code (https://docs.github.com/en/github/collaborating-with-issues-and-pull-requests/resolving-a-merge-conflict-on-github#:~:text=You%20can%20only%20resolve%20merge,locally%20on%20the%20command%20line.)

TheTrueProblematic commented:
> Your configure.ac looks good, and you are now passing CI, which is good.
> 
> (Except travis. I will take a look...)
> 
> Python code does not have to be compiled, so all we have to do is make sure the file ends up in the tarball distribution. (https://www.gnu.org/software/automake/manual/html_node/Dist.html). We need an EXTRA_DIST = filename in the Makefile.am.
> 
> In the ptest directory we need to run the tests. For that, we need a TESTS = testname in the Makefile.am (this is documented, in horrifying complexity, here https://www.gnu.org/software/automake/manual/html_node/Parallel-Test-Harness.html. But it's just gong to be one line: TESTS = script.py)

I have a couple questions regarding this. 
First, Travis is showing as running for me, is it different between us?
Second, which Makefile.am need the EXTRA_DIST = filename in it? 
Thirdly, for ptest, is the idea to write tests in the python files and then call them from the TESTS = testname in the Makefile.am of the ptest directory? 
Thanks for the help!

edhartnett commented:
Makefile.am in psrc needs EXTRA_DIST.

Makefile.am needs TESTS target.

On Tue, Mar 16, 2021, 4:50 PM TheTrueProblematic ***@***.***>
wrote:

> Your configure.ac looks good, and you are now passing CI, which is good.
>
> (Except travis. I will take a look...)
>
> Python code does not have to be compiled, so all we have to do is make
> sure the file ends up in the tarball distribution. (
> https://www.gnu.org/software/automake/manual/html_node/Dist.html). We
> need an EXTRA_DIST = filename in the Makefile.am.
>
> In the ptest directory we need to run the tests. For that, we need a TESTS
> = testname in the Makefile.am (this is documented, in horrifying
> complexity, here
> https://www.gnu.org/software/automake/manual/html_node/Parallel-Test-Harness.html.
> But it's just gong to be one line: TESTS = script.py)
>
> I have a couple questions regarding this.
> First, Travis is showing as running for me, is it different between us?
> Second, which Makefile.am need the EXTRA_DIST = filename in it?
> Thirdly, for ptest, is the idea to write tests in the python files and
> then call them from the TESTS = testname in the Makefile.am of the ptest
> directory?
> Thanks for the help!
>
> —
> You are receiving this because you commented.
> Reply to this email directly, view it on GitHub
> <https://github.com/edhartnett/ncglm/pull/30#issuecomment-800664157>, or
> unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABCSXXE5GDIE3VDAWLH2KK3TD7OEBANCNFSM4XZHPT7A>
> .
>

TheTrueProblematic committed:
Added EXTRA_DIST

TheTrueProblematic committed:
Pointed to test file

edhartnett commented:
@TheTrueProblematic can you resolve the conflicts and we can merge this and move on to next steps?

TheTrueProblematic committed:
New work on testing file!

edhartnett committed:
Merge branch 'master' into mwm-wrapperstart
